### PR TITLE
fix(mount): check if path is mounted and create /run/initramfs/live if doesn't exist

### PR DESCRIPTION
**Pull request by mudler**
Attempt to fix https://github.com/kairos-io/kairos/issues/2224

Closes https://github.com/kairos-io/kairos/issues/2224.

### CODE REVIEW

File: pkg/utils/partitions/getpartitions.go [hunk 1]
```diff
@@ -209,3 +209,24 @@ func GetPartitionViaDM(fs v1.FS, label string) *v1.Partition {
 	}
 	return part
 }
+
+func IsPathMounted(fs v1.FS, s string) (bool, error) {
```
Comment (Itxaka) [inline pkg/utils/partitions/getpartitions.go:213]: dont we have a util from the mount lib that checks this? Maybe we dont need to reimplement it already?

Comment (Itxaka) [reply pkg/utils/partitions/getpartitions.go:213]: yep, there is a `IsNotMountPoint` from the mounter, maybe we can reuse it?

```
// IsNotMountPoint determines if a directory is a mountpoint.
// It should return ErrNotExist when the directory does not exist.
// IsNotMountPoint is more expensive than IsLikelyNotMountPoint.
// IsNotMountPoint detects bind mounts in linux.
// IsNotMountPoint enumerates all the mountpoints using List() and
// the list of mountpoints may be large, then it uses
// isMountPointMatch to evaluate whether the directory is a mountpoint.
func IsNotMountPoint(mounter Interface, file string) (bool, error) {
```

Comment (Itxaka) [reply pkg/utils/partitions/getpartitions.go:213]: You also got `i.cfg.Mounter.IsLikelyNotMountPoint(constants.UkiCdromSource)` available but we are using a fakeMounter so we need to set a proper mounter on config build for that to work.

Whatever you wish, but thats the mounter check from kubernetes so I would say that has been thoroughly tested lol

**Review by Itxaka — APPROVED** [review]

File: pkg/utils/partitions/getpartitions.go (hunk 1 above)
Comment (mudler) [reply pkg/utils/partitions/getpartitions.go]: > You also got `i.cfg.Mounter.IsLikelyNotMountPoint(constants.UkiCdromSource)` available but we are using a fakeMounter so we need to set a proper mounter on config build for that to work.
> 
> Whatever you wish, but thats the mounter check from kubernetes so I would say that has been thoroughly tested lol

alright alright, moving to using IsMountPoint() then :)

File: pkg/uki/install.go [hunk 2]
```diff
@@ -50,10 +56,14 @@ func (i *InstallAction) Run() (err error) {
 		MountPoint: constants.UkiSource,
 	}
 
-	err = e.MountImage(image)
-	if err != nil {
-		return err
+	mounted, err := i.cfg.Mounter.IsMountPoint(constants.UkiCdromSource)
```
Comment (Itxaka) [inline pkg/uki/install.go:59]: argh shit, sorry this wont work. The mounter in here is a FakeMounter as we didnt needed a real mounter, soi this check is done in memory mounts only, not real ones....

Comment (Itxaka) [reply pkg/uki/install.go:59]: or not, fuck let me check Im a bit confused today

Comment (Itxaka) [reply pkg/uki/install.go:59]: yeah Im stupid, this is good, I was confusing this with enki.

File: pkg/uki/install.go [hunk 3]
```diff
@@ -35,8 +36,13 @@ func (i *InstallAction) Run() (err error) {
 
 	// immucore mounts CDROM under this path
 	_, err = i.cfg.Fs.Stat(constants.UkiCdromSource)
-	if err != nil {
-		i.cfg.Logger.Errorf("Error checking if %s exists: %s", constants.UkiCdromSource, err)
+	if errors.Is(err, os.ErrNotExist) {
+		// The directory is under /run, so we can create it if not present
+		if err := fsutils.MkdirAll(i.cfg.Fs, constants.UkiCdromSource, os.ModeDir|os.ModePerm); err != nil {
```
Comment (Itxaka) [inline pkg/uki/install.go:41]: no, this is wrong, if that path doesnt exists it means that immucore didnt mount the cdrom under there, so the next step wont work. We NEED that path mounted to be able to mount the following image, creating the dir here only hides the issue.

**Pull request merged**